raiden/waiting.py

```diff
@@ -149,27 +149,43 @@ def wait_for_channel_in_states(
 ) -> None:
     """Wait until all channels are in `target_states`.
 
+    Raises:
+        ValueError: If the token_address is not registered in the
+            payment_network.
+
     Note:
         This does not time out, use gevent.Timeout.
     """
-    channel_ids = list(channel_ids)
+    chain_state = views.state_from_raiden(raiden)
+    token_network = views.get_token_network_by_token_address(
+        chain_state=chain_state,
+        payment_network_id=payment_network_id,
+        token_address=token_address,
+    )
 
-    while channel_ids:
-        last_id = channel_ids[-1]
+    if token_network is None:
+        raise ValueError(
+            f'The token {token_address} is not registered on the network {payment_network_id}.',
+        )
+
+    token_network_address = token_network.address
+
+    list_cannonical_ids = [
+        CanonicalIdentifier(
+            chain_identifier=chain_state.chain_id,
+            token_network_address=token_network_address,
+            channel_identifier=channel_identifier,
+        )
+        for channel_identifier in channel_ids
+    ]
+
+    while list_cannonical_ids:
+        canonical_id = list_cannonical_ids[-1]
         chain_state = views.state_from_raiden(raiden)
-        token_network_address = views.get_token_network_by_token_address(
-            chain_state=chain_state,
-            payment_network_id=payment_network_id,
-            token_address=token_address,
-        ).address
-        assert token_network_address
```

Comment:
This assert was just plain wrong, it would never be reached because the problem would manifest itself as `None` being the return value from `get_token_network_by_token_address`, and on the line above it would fail with `None has no attribute address`